lib/rules/no-unused-vars.js

```diff
@@ -613,6 +618,14 @@ module.exports = {
                             continue;
                         }
 
+                        if (type === "ClassName") {
+                            const hasStaticBlock = def.node.body.body.filter(node => node.type === "StaticBlock");
```

Comment:
```suggestion
                            const hasStaticBlock = def.node.body.body.some(node => node.type === "StaticBlock");
```

`filter` always returns an array, so this would always be truthy.

```js
/* eslint no-unused-vars: ["error", { ignoreClassWithStaticInitBlock: true }] */

class C {} // false negative
```

We should also add some tests where this option is `true` and class doesn't have static blocks.

Reply:
Done!